src/west/main.py

```diff
@@ -517,7 +517,9 @@ def main(argv=None):
         log.dbg("sys.executable:\"{}\" sys.argv[0]:\"{}\" argv:\"{}\"".format(
                 sys.executable, sys.argv[0], argv))
         # Treat the Python script as an executable
-        os.execv(sys.argv[0], argv)
+        proc = Popen([sys.argv[0]] + argv)
```

Comment:
Does this break control-C in a gdb started by `west debug`? Attempts to run west as a subprocess generally do, because the SIGINT gets delivered to the parent process, which exits, instead of the subprocess west, which is temporarily ignoring it so GDB can handle it.

Reply:
It might have,  but `west debug` will never raise the exception. Only  commands that update repos and the help command can raise it. So no interactive commands raise the exception.

Reply:
To add to this, we will remove the `selfupdate` when we move everything to pypi and also the current `execv` is actually broken on Windows, there's a race and sometimes you can end up with having to press enter to return to the shell. Finally, I think it's more likely that control-C will work reliably with `Popen` than with `execv`, at least on Windows

Reply:
Re: control-C you can see that actually both child and parent get the signal [here](https://stackoverflow.com/a/19807536/337201)